modules/core/nuxeo-platform-commandline-executor/src/test/java/org/nuxeo/ecm/platform/commandline/executor/tests/TestCommands.java

```diff
@@ -119,4 +122,43 @@ public void testCmdPipe() throws Exception {
         assertEquals(expected, line);
     }
 
+
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.commandline.executor:OSGI-INF/commandline-command-test-contrib.xml")
+    public void testTimeoutOnBlockingCommand() throws Exception {
+        Assume.assumeTrue("Requires timeout", cles.getAvailableCommands().contains("timeout"));
+        List<String> cmds = cles.getRegistredCommands();
+        assertTrue(cmds.contains("block"));
+
+        ExecResult result = cles.execCommand("block", cles.getDefaultCmdParameters());
+        assertFalse(result.isSuccessful());
+        // Command terminated with SIGKILL = 128 + 9
+        assertEquals(137, result.getReturnCode());
+        assertTrue(result.isCommandInTimeout());
+    }
+
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.commandline.executor:OSGI-INF/commandline-command-test-contrib.xml")
+    public void testTimeoutOnTransaction() throws Exception {
+        Assume.assumeTrue("Requires timeout", cles.getAvailableCommands().contains("timeout"));
+        List<String> cmds = cles.getRegistredCommands();
+        assertTrue(cmds.contains("tooLong"));
+
+        // Execute a long command
+        ExecResult result = cles.execCommand("tooLong", cles.getDefaultCmdParameters());
+        assertTrue(result.isSuccessful());
+        assertFalse(result.isCommandInTimeout());
+
+        // Execute the command again but within a short transaction
+        TransactionHelper.commitOrRollbackTransaction();
+        TransactionHelper.startTransaction(1);
+        result = cles.execCommand("tooLong", cles.getDefaultCmdParameters());
+        assertFalse(result.isSuccessful());
+        assertTrue(result.isCommandInTimeout());
```

Comment:
Don't we want to test the return code, as in the test above?

Reply:
No need, this is done in isCommandInTimeout.